WTG.Analyzers.Utils/ExpressionSyntaxFactory.cs

```diff
@@ -7,6 +8,25 @@ namespace WTG.Analyzers.Utils
 {
 	public static class ExpressionSyntaxFactory
 	{
+		public static IdentifierNameSyntax Nameof
+		{
+			get
+			{
+				if (nameofSyntax == null)
+				{
+					// This may seem stupid (lets face it, it is) but the identifier produced by SyntaxFactory.IdentifierName
+					// has the wrong kind of magic for the compiler to recognise it as a contextual keyword.
+					//
+					// Error CS0103: The name 'nameof' does not exist in the current context
+					var invoke = (InvocationExpressionSyntax)SyntaxFactory.ParseExpression("nameof(A)");
```

Comment:
Surely there has to be a better way to do this?

Reply:
If there is, I have not been able to find it.

I have noticed two differences in the identifier token
1. the identifier token produced by `SyntaxFactory.IdentifierName` has an annotation which isn't there when parsing (which I have not yet been able to identify).
2. the tokens underlying 'green' node has a `contextualKind` of `NameOfKeyword` when parsed in this way and `IdentifierToken` when produced by `SyntaxFactory.IdentifierName`. (The token kind is still `IdentifierToken` in both cases)

Reply:
I'm not holding out much hope that there will be an alternative in the near future. https://github.com/dotnet/roslyn/issues/24212